share/getters/cascade.go

```diff
@@ -122,16 +121,18 @@ func cascadeGetters[V any](
 		getCtx, cancel := ctxWithSplitTimeout(ctx, len(getters)-i, 0)
 		val, getErr := get(getCtx, getter)
 		cancel()
-		if getErr == nil {
-			return val, nil
+		if getErr == nil || errors.Is(getErr, share.ErrNamespaceNotFound) {
+			return val, getErr
 		}
-		if errors.Is(share.ErrNamespaceNotFound, getErr) {
-			return zero, getErr
+
+		if errors.Is(getErr, errOperationNotSupported) {
+			continue
 		}
 
-		if !errors.Is(getErr, errOperationNotSupported) {
-			err = errors.Join(err, getErr)
-			span.RecordError(getErr, trace.WithAttributes(attribute.Int("getter_idx", i)))
+		err = errors.Join(err, getErr)
+		span.RecordError(getErr, trace.WithAttributes(attribute.Int("getter_idx", i)))
+		if ctx.Err() != nil {
```

Comment:
Let's do this check right after the call. If user cancelled, there is no point to go further.

Reply:
Ok, if you want to return aggregated error, then it's ok to leave it like this